library/Data/Field.hs

```diff
@@ -248,18 +234,14 @@ instance IsNegative Rational
 deriving via ABaseField (GF.Prime p) instance KnownNat p => Field (GF.Prime p)
 
 instance KnownNat p => AsInteger (GF.Prime p) where
-    asInteger = Just . GF.fromP
-
-instance KnownNat p => IsNegative (GF.Prime p) where
-    isNegative f = i > p `Prelude.div` 2 where
-        p = natVal @p Proxy
-        i = GF.fromP f
+    asInteger x
+        | i < p `Prelude.div` 2 = Just i
```

Comment:
Does the edge case of ``i == p `Prelude.div` 2`` matter here. I am not sure.

Reply:
I started checking that and realized the whole test is broken and doesn't test what it's supposed to. Will fix.